luigi_pipeline/lib/model/sv_mt_schema.py

```diff
@@ -0,0 +1,217 @@
+            hl.is_defined,
+            mapped_genes
+        ).flatmap(lambda x: x)
+
+    @row_annotation(fn_require=end_locus)
+    def xstop(self):
+        return variant_id.get_expr_for_xpos(self.mt.end_locus)
+
+    # rg37_locus is annotated by HailMatrixTableTask.add_37_coordinates
+
+    @row_annotation(fn_require=end_locus)
+    def rg37_locus_end(self):
+        return hl.or_missing(
+            self.mt.end_locus.position <= hl.literal(hl.get_reference('GRCh38').lengths)[self.mt.end_locus.contig],
+            hl.liftover(hl.locus(self.mt.end_locus.contig, self.mt.end_locus.position, reference_genome='GRCh38'), 'GRCh37'),
+        )
+
+    @row_annotation(name='svType')
+    def sv_type(self):
+        return self.sv_types()[0]
+
+    @row_annotation(name='transcriptConsequenceTerms', fn_require=[
+        sorted_transcript_consequences, sv_type,
+    ])
+    def transcript_consequence_terms(self):
+        return self.mt.sortedTranscriptConsequences.map(lambda x: x[MAJOR_CONSEQUENCE]).extend([self.mt.svType])
+
+    @row_annotation()
+    def sv_type_detail(self):
+        sv_types = self.sv_types()
+        return hl.if_else(
+            sv_types[0] == 'CPX',
+            self.mt.info.CPX_TYPE,
+            hl.or_missing(
+                (sv_types[0] == 'INS') & (hl.len(sv_types) > 1),
+                sv_types[1],
+            )
+        )
+
+    @row_annotation(name='geneIds', fn_require=sorted_transcript_consequences)
+    def gene_ids(self):
+        return hl.set(
+            self.mt.sortedTranscriptConsequences.filter(
+                lambda x: x[MAJOR_CONSEQUENCE] != 'NEAREST_TSS'
+            ).map(
+                lambda x: x[GENE_ID]
+            )
+        )
+
+    @row_annotation(name='variantId', disable_index=True)
+    def variant_id(self):
+        return self.mt.rsid
+
+    # NB: This is the "elasticsearch_mapping_id" used inside of export_table_to_elasticsearch.
+    @row_annotation(name='docId', disable_index=True)
+    def doc_id(self, max_length=512):
+        return self.mt.rsid[0: max_length]
+
+
+class SeqrSVGenotypesSchema(SeqrGenotypesSchema):
+
+    def _genotype_fields(self):
+        is_called = hl.is_defined(self.mt.GT)
+        return {
+            'sample_id': self.mt.s,
+            'gq': self.mt.GQ,
+            'cn': self.mt.RD_CN,
+            'num_alt': hl.if_else(is_called, self.mt.GT.n_alt_alleles(), -1)
+        }
+
+    # NB: override this function here to mimic the existing null handling behavior.
+    def _genotype_filter_samples(self, filter_):
+        samples = super()._genotype_filter_samples(filter_)
+        return hl.or_missing(hl.len(samples) > 0, samples)            
+
+    @row_annotation(name="samples_gq_sv", fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_gq(self):
+        # NB: super().samples_gq is a RowAnnotation... so we call the method under the hood.
+        # ew it is gross.
+        return super().samples_gq.fn(self, start=0, end=1000, step=10)
```

Comment:
would the change you made to make RowAnnotation callable mean you could do `super().samples_gq(self, start=0, end=1000, step=10)` or no?

Reply:
I think it would work if we changed the `__call__` to take in **kwargs which would then be passed to the function.  That might be cleaner?

Reply:
Tried it and it doesn't work.  Calling the RowAnnotation doesn't return the function result but returns the instance of the schema class... which is very unintuitive.

There were a couple of tests doing things like:

```
test_schema.a().b().c_1()
```
and 
```
mt = test_schema.d().select_annotated_mt()
```

I'm going to leave this for now and we can revisit.